loan-calculator-client/src/components/MediaPortal.js

```diff
@@ -0,0 +1,465 @@
+            callerCandidatesCollection.add(event.candidate.toJSON());
+        });
+
+        //Step 5: Functionality to create a room:
+        const offer = await peerConnection.createOffer()
+        await peerConnection.setLocalDescription(offer)
+        console.log('Created offer:', offer)
+
+        const roomWithOffer = {
+            'offer': {
+                type: offer.type,
+                sdp: offer.sdp,
+            }
+        }
+        await roomRef.set(roomWithOffer);
+
+        //update the admin user with their own private room channel
+        let user = firebase.auth().currentUser
+        if (user) {
+            let adminUsersRef = await db.collection('admin-users').doc(user.uid)
+
+            //Update the admin_video_room
+            adminUsersRef.update({
+                "admin_video_room": roomRef.id
+            })
+            .then(() => {
+                this.setState({roomId: roomRef.id}, async function() {
+                    console.log(`New room created with SDP offer. Room ID: ${roomRef.id}`);
+
+                    //Send this room as a message over to the other user to join
+                    adminUsersRef.get().then((doc) => {
+                        if (doc.exists) {
+                            let messagesRef = db.collection('chat-rooms')
+                                                .doc(doc.data().admin_room_location)
+                                                .collection('messages')
+                                                .doc()
+                            let anonUsersRef = db.collection('anon-users')
+                                                 .doc(doc.data().admin_room_location)
+                            //Update the anonymous user's video room to match the admin's video room
+                            anonUsersRef.update({"anon_video_room": roomRef.id})
+                                        .then(function() {
+                                            console.log("Anonymous User's Video Room matches Admin User's Video Room")
+                                            alert("Successfully Created Video Room!")
+                                         }).catch(function(err) {
+                                            console.log(err)
+                                         })
+                            messagesRef.set({
+                                "sender_name": doc.data().admin_name,
+                                "message": "Click the \"Join Room Button\" to join my video call.",
+                                "uid": user.uid,
+                                "timestamp": firebase.firestore.FieldValue.serverTimestamp()
+                            })
+                        } else {
+                            console.log("no such document")
+                        }
+                    }).catch((err) => console.log(err))
+                })                
+            })
+            .catch((err) => console.log(err))
```

Comment:
Not a huge deal, especially since we only have a couple days left in the sprint, but I generally think handling errors should be more than just printing them because we wouldn't want someone using our app to see it crash.

Reply:
I agree 100%. I came across the **componentDidCatch** lifecycle method and will add that to the code as well.